python/ray/autoscaler/aws/node_provider.py

```diff
@@ -67,7 +67,12 @@ def __init__(self, provider_config, cluster_name):
         # Try availability zones round-robin, starting from random offset
         self.subnet_idx = random.randint(0, 100)
 
-        self.tag_cache = {}  # Tags that we believe to actually be on EC2.
+        # Cache of node objects from the last nodes() call. This avoids
+        # excessive DescribeInstances requests.
+        self.provider_cache = provider_cache
+        if provider_cache == None:
+            self.provider_cache = NodeProviderCache()
+
```

Comment:
Can we replace this with
```
self.provider_cache = provider_cache or NodeProviderCache()
```
throughout the code?

Reply:
okay.